cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/BasePagingService.cs

```diff
@@ -25,10 +34,13 @@ public virtual async Task<Stream> GetPagedDataAsync(Func<RequestInformation, Can
         Stream? response = null;
         do
         {
-            var pageData = await requestExecutorAsync(requestInfo, cancellationToken);
+            var responseHandler = CreateResponseHandler();
+            await requestExecutorAsync(requestInfo, responseHandler, cancellationToken);
+            var pageData = await ExtractResponseStreamAsync(responseHandler, cancellationToken);
+            var headers = ExtractResponseHeaders(responseHandler);
```

Comment:
Related to the first comment. 
Maybe we should just extract the stream and headers directly from the `HttpResponseMessage` if there are no other scenarios we were looking to support?

```suggestion
            var responseHandler = new NativeResponseHandler();
            await requestExecutorAsync(requestInfo, responseHandler, cancellationToken);
            var responseMessage = responseHandler.Value as HttpResponseMessage;
            var pageData = await responseMessage?.Content.ReadAsStreamAsync(cancellationToken);
            var headers = responseMessage.Headers.ToDictionary(x => x.Key,x => x.Value);
```

Reply:
See earlier comment.